src/rt/mod.rs

```diff
@@ -21,6 +20,34 @@ pub mod marker;
 
 pub use wasm_bindgen_macro::BindgenedStruct;
 
+// This macro is a hack to implement "generic" casts and reduce number of
+// boilerplate intrinsics. The implementation generates a no-op JS adapter that
+// simply takes an argument in one type, decodes it from the ABI, does nothing
+// with it on the JS side (by declaring function name as empty, so instead of
+// generating typical JS call that does `ret = foo(arg);` we end up with
+// just `ret = (arg);` and then encoding same value back with a different type.
+//
+// Note that we intentionally keep this macro undocumented as it's not meant
+// for public consumption - it relies on our implementation details, and
+// is intended only for simplifying our own implementation details as well.
+//
```

Comment:
```suggestion
```
Everything in the `rt` module isn't meant to be public and is `doc(hidden)` already.

Reply:
Unfortunately, that doesn't apply to `macro_rules!`. They don't behave by normal scoping rules and always end up in the global scope of the crate, thus visible even if the module is not. One of my personal pet peeves for many years 😅

Reply:
Yes, right, it still requires `doc(hidden)`. What I was trying to say is that we have a bunch of macros inside the `rt` module, we don't need to tell developers on every macro that this isn't for public consumption.

So `rt` implies its not for public consumption so it shouldn't be here otherwise. The comment confused me for that reason.

Reply:
Ah ok. FWIW I only moved it to this module a few minutes ago after Guy's review above. Happy to shorten the comment.